### PR TITLE
Add WordMatcher

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: src/WordMatcher.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    /**
+     * @param string $needle
+     *
+     * @return null|string
+     */
+    public function match($needle)
+    {
+        $word = null;
+        $distance = 7;
```
Comment (julienfalque) [inline src/WordMatcher.php]: `levenshtein('braces', 'elseif') // 6`

Maybe this value should be computed from the length of the compared words?

Comment (keradus) [reply src/WordMatcher.php]: I tried to adjust it with way how you proposed
I have feeling that this value will be magic to determine ;)

File: src/WordMatcher.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * @author SpacePossum
+ *
+ * @internal
+ */
+final class WordMatcher
+{
+    /**
+     * @var string[]
+     */
+    private $dict;
```
Comment (julienfalque) [inline src/WordMatcher.php]: `$dict` is a bit confusing IMO, why not simply `$candidates`?

Comment (keradus) [reply src/WordMatcher.php]: both good, changing

File: src/WordMatcher.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+/**
+ * Base class for function reference fixers.
```
Comment (julienfalque) [inline src/WordMatcher.php]: Description looks wrong.

File: tests/WordMatcherTest.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+final class WordMatcherTest extends TestCase
+{
+    /**
+     * @param null|string $expected
+     * @param string      $needle
+     * @param array       $dict
+     *
+     * @dataProvider provideMatchCases
+     */
+    public function testMatch($expected, $needle, array $dict)
```
Comment (julienfalque) [inline tests/WordMatcherTest.php]: I would rename `$dict` here as well.

**Pull request closed without merging**